src/files/utils.ts

```
@@ -0,0 +1,96 @@
+    lastModifiedDate: lastModifiedDateHeader
+      ? new Date(lastModifiedDateHeader)
+      : new Date(),
+  };
+
+  if (!metadata) {
+    return defaultFileData;
+  }
+
+  return metadata.reduce((file, metadataKey) => {
+    return {
+      ...file,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)